[Path] Fix outside `Extensions` of circular faces with inner holes

Russ4262 opened the pull request:
Fix bug identified by MLampert.  This fix addresses extensions for circular faces that have non-extended holes in the face.
.
Unlike polygon extensions that only need one wire to define the extension shape, some circular extensions require two wires (a face would be better) to define the extension area.  

A part of the problem is that the current `Extensions` code returns an extension area using a single wire; however, some extension areas need two wires to be correctly defined.  This fix makes use of a new method to identify the circular extension region using a `Part.Face()` shape to allow for the two wires that are necessary to define the circular extension.

Forum discussion at [PR fixes for PocketShape: 4615, 4617](https://forum.freecadweb.org/viewtopic.php?f=15&t=56667).  Bug was identified earlier today in a Gitter chat.

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [x]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [x]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [x]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [x]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the [FreeCAD bug tracker issue number](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) in case a particular commit solves or is related to an existing issue on the tracker. Ex: `Draft: fix typos - fixes #0004805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.19 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=34586).

---

mlampert reviewed (approved):
Looks good, would be great to have a unit test for this case ;)

Russ4262 commented:
I just added a fix for the GUI visualization for the circular-face extensions.  I also rebased this PR.  The recent commit corrects the visualization to show a the circular extension ring rather than the incorrect complete circle.

Russ4262 commented:
I restructured the code a bit to accommodate future improvements to the `Extensions` code that might require multiple wires to define the extension area.
